src/vuln_analysis/utils/source_rpm_downloader.py

```diff
@@ -0,0 +1,731 @@
+        """
+        Download RPMs in parallel using threads. Each SBOM entry is downloaded in a separate thread.
+        """
+        logger.info(f"Starting threaded RPM download with {max_workers} workers...")
+        # Use singleton's cache directory instead of hardcoded value
+        rpm_cache_dir = self.rpm_cache_dir
+        downloader = SourceRPMDownloader(
+            project_name, sbom, self.base_urls,
+            download_path, rpm_cache_dir, max_workers
+        )
+        downloader.run()        
+
+    # def build_dependency_tree(self, root_path):
+    #    logger.info(f"Building dependency tree for {root_path} ...")
+    #    builder = CCppDependencyTreeBuilder()
+        # Add common include paths if they exist
+    #    source_path_obj = Path(root_path)
+    #    for subdir in ["include", "src"]:
+    #        candidate = source_path_obj / subdir
+    #        if candidate.exists():
+    #            builder.add_include_path(str(candidate))
+    #    start_time = time.time()
+        # Use multi-threaded mode (max_workers=4)
+    #    tree, cache_stats = builder.build_dependency_tree_enhanced(source_path_obj, max_workers=4)
+    #    elapsed = time.time() - start_time
+    #    logger.info(f"Dependency tree built with {len(tree)} nodes in {elapsed:.2f} seconds.")
+        
+    #    return tree, elapsed, cache_stats
+
+   # def prune_unused_rpms(self, dependency_tree, project_name, rpm_libs_path, depth=4):
+   #     import shutil
+   #     from collections import deque
+   #     # BFS to depth
+   #     needed = set()
+   #     queue = deque([(project_name, 0)])
+   #     while queue:
+   #         node, d = queue.popleft()
+   #         if node in needed or d > depth:
+   #             continue
+   #         needed.add(node)
+   #         if node in dependency_tree:
+   #             for child in dependency_tree[node]:
+   #                 queue.append((child, d + 1))
+   #     # List all RPM dirs
+   #     rpm_libs = Path(rpm_libs_path)
+   #     all_rpm_dirs = [p.name for p in rpm_libs.iterdir() if p.is_dir()]
+        # Prune: keep only those in needed
+   #     to_delete = [d for d in all_rpm_dirs if d not in needed]        
+   #     logger.info(f"RPMs that can be deleted (not needed for analysis): {len(to_delete)} from total {len(all_rpm_dirs)}")
+        
+   #     # Delete unused RPM directories
+   #     for d in to_delete:
+   #         dir_path = rpm_libs / d
+   #         try:
+   #             shutil.rmtree(dir_path)                
+   #         except Exception as e:
+   #             logger.error(f"Failed to delete {dir_path}: {e}")
+        # Remove deleted elements from the dependency tree
+   #     updated_tree = {k: [v for v in vs if v not in to_delete] for k, vs in dependency_tree.items() if k not in to_delete}
+   #     return updated_tree
+
+   # def write_output(self, output_path, data):
+   #     with open(output_path, 'w') as f:
+   #         json.dump(data, f, indent=2)
```

Comment:
@RedTanny Please remove with informative meaningful commit ( no need to keep it commented in the code, as it's to much verbose, and we need to keep the code clean as much as possible), if needed to lookup on that as reference later , you can lookup on this comments at the "removing commit" in the history.
```suggestion
```